scrapy/http/request/form.py

```diff
@@ -120,7 +115,7 @@ def _get_form(
     formxpath: Optional[str],
 ) -> FormElement:
     """Find the wanted form element within the given response."""
-    root = create_root_node(response.text, HTMLParser, base_url=get_base_url(response))
+    root = response.selector.root
```

Comment:
Can you please explain this? Also does this work with all supported parsel versions?

Reply:
Some of the code later on relies on attributes from `lxml.etree.Element` not exposed by `Selector` (e.g. `name`), so using the selector all the way was not possible.

[`Selector.root` was already available in the lowest parsel version Scrapy supports](https://github.com/scrapy/parsel/blob/v1.5.0/parsel/selector.py#L200).